### PR TITLE
Implement CI checks and linter

### LINKED ISSUE #75: CI Checks and Linter Addition
**Is your feature request related to a problem? Please describe.**
The project lacks PR checks where there are no requirements and no baseline checks for accepting them, which could lead to unintentional merges.

**Describe the solution you'd like**
Setup a CI tool (Travis, Circle) based on what works best with the project, ensuring the availability of a linter on top of it to ensure that the best practices are being followed.

### PR DESCRIPTION
**Pull request by rejahrehim**
This PR covers issue #<issue number>.

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Implements Travis to lint the files before merging 
- closes #75

### CODE REVIEW

**Comment by rejahrehim** [comment]
Need to override the test to implement it for the first time.

File: README.md [hunk 1]
```diff
@@ -8,20 +8,21 @@ You can download the stable version v4 [here](http://www.owasp.org/index.php/OWA
 
 ## Discussion/Support
 
-Join the Testing Guide Google Group if there are things you want to discuss https://groups.google.com/a/owasp.org/forum/#!forum/testing-guide-project
+Join the Testing Guide Google Group if there are things you want to discuss <https://groups.google.com/a/owasp.org/forum/#!forum/testing-guide-project>
```
Comment (kingthorin) [inline README.md]: This doesn't seem like github flavoured markdown (GFM)

- https://guides.github.com/features/mastering-markdown/
- https://help.github.com/en/articles/basic-writing-and-formatting-syntax
- https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

Comment (rejahrehim) [reply README.md]: Updated

**Comment by patrickceg** [comment]
I'm a Bitbucket / Jenkins person, so can you describe what is needed to run the checker? Does it run on its own automatically for pull requests, or can you manually invoke the checks (i.e. on your local machine)?

**Comment by patrickceg** [comment]
Can you describe what is needed to run the checker? Does it run on its own automatically for pull requests, or can you manually invoke the checks (i.e. on your local machine)?

I think we just need a one-liner in the contribution guide. Something like:

- Verify your change passes the automated linter check [link to tutorial]

**Comment by ThunderSon** [comment]
@patrickceg it runs automatically on any PR or PR update/change.

**Comment by rejahrehim** [comment]
> Can you describe what is needed to run the checker? Does it run on its own automatically for pull requests, or can you manually invoke the checks (i.e. on your local machine)?
It runs automatically on every PR. Also if anyone needs to configure it locally just install the node modules and run npm test. 

npm install

npm test

**Comment by kingthorin** [comment]
Could we add:
```text
# Eclipse
# -------
*.classpath
*.project
*.settings
```

to `.gitignore`? Actually what do you think about splitting .gitignore into it's own PR and getting that done sooner than later?

**Comment by rejahrehim** [comment]
Okay. I will update and create another PR for gitignore

**Comment by rejahrehim** [comment]
Moved gitignore to another PR #109

**Comment by ThunderSon** [comment]
From the package file, I can see that we are still relying on remark lint, and not markdown lint, right? Might we be able to switch in order to have the same checks as the development environment, and what are your thoughts on the matter?

**Comment by rejahrehim** [comment]
Both are almost similar (https://github.com/remarkjs/remark-lint/blob/master/doc/comparison-to-markdownlint.md). Remark lint is much stable and the output seems good. Running markdownlint-cli is a bit tricky (need to write test script using glob and husky). In our case, we need to implement some custom rules too. I have implemented those rules with remark-cli.

**Comment by rejahrehim** [comment]
Run using markdownlint-cli
Screenshot 2019-07-23 at 11 32 47 PM

Run using remark-cli
Screenshot 2019-07-23 at 11 34 44 PM

**Comment by rejahrehim** [comment]
@ThunderSon Need to check this PR

**Comment by kingthorin** [comment]
This PR won't get "happy" until we get through the migration activity(ies).

**Comment by kingthorin** [comment]
Oops?

**Comment by rejahrehim** [comment]
Sorry, My Push closed it.

**Comment by rejahrehim** [comment]
Updated the linter to markdownlint and fixed the conflicts. I will put another PR.

**Pull request closed without merging**